packages/packages.go

```diff
@@ -81,7 +81,7 @@ type Packages struct {
 
 // PkgInfo describes a package.
 type PkgInfo struct {
-	Name, Arch, Version string
+	Name, Arch, Version, NameWithArch string
```

Comment:
Why we need NameWithArch if we already have Name and Arch in the structure?

Reply:
I suggest to replace Arch with a specific type rather than just string, it might store both raw and standardized architecture, that will help us to avoid create new field that is required only to specific use-case for specific package manager.

Reply:
The already existing Arch is a little different, it resolves the different architecture this way (https://github.com/GoogleCloudPlatform/osconfig/blob/a235d3366ebc7183a106d62dddd22cfe1f18822d/osinfo/osinfo.go#L31) while reporting to inventory. 

I am not comfortable changing this for our existing reporting, hence the new field.

Reply:
And manual testing by me verified that "yum install foo.noarch" succeeds whereas the resolved "yum install foo.all" will fail here.